compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -233,21 +233,24 @@ class TreeUnpickler(reader: TastyReader,
 // ------ Reading types -----------------------------------------------------
 
     /** Read names in an interleaved sequence of types/bounds and (parameter) names,
-     *  possibly followed by a sequence of modifiers.
+     *  possibly followed by a sequence of modifiers and a list of erased parameter flags.
      */
-    def readParamNamesAndMods(end: Addr): (List[Name], FlagSet) =
+    def readParamNamesAndMods(end: Addr): (List[Name], FlagSet, List[Boolean]) =
```

Comment:
I wonder if we should future-proof method types and make an encoding that allows a modifier for each parameter.

Reply:
I think it would be nice, though I wonder what's the best way to do it without a lot of extra bytes needed for each parameter, even if they don't have any modifiers...